### ISSUE TITLE
Is this compatible with the flutter freezed package?

### ISSUE THREAD

**Issue by BenjaminFarquhar** [smotastic/smartstruct#29]
I haven't managed to map my [freezed](https://pub.dev/packages/freezed) objects successfully. Is this possible at this stage?

**Comment by andrew-tpz** [comment]
Hi, any progress with that?

I use smartstruct v1.2.4+1 and freezed: v1.0.2+1.
Mapper file is generated withput errors, but generated code is wrong, because Person is immutable.

```
class PersonMapperImpl extends PersonMapper {
  PersonMapperImpl() : super();

  @override
  Person fromModel(PersonModel model) {
    final person = Person();
    person.name = model.name;
    person.age = model.age;
    person.copyWith = model.copyWith;
    return person;
  }
}
```

**Comment by smotastic** [comment]
Hi,
I merged a fix (#47), so you should be able to use freezed with smartstruct now (in the next release which will be coming soon).
As @andrew-tpz noted, the wrong constructor was used, so now the constructor with the most parameters is used. In the case of freezed, one of the given factory constructors.

Another problem was the mapping of the copyWith Method, this has to be ignored manually.
See the example on how to use freezed (https://github.com/smotastic/smartstruct/tree/develop/example/lib/freezed) or the README.

As I didn't use freezed much, I hope most of it will work now.
If you find any more problems please feel free to open another issue.
